glean-core/src/lib_unit_tests.rs

```diff
@@ -812,6 +815,158 @@ fn test_setting_log_pings() {
     assert!(!glean.log_pings());
 }
 
+#[test]
+fn test_set_metrics_disabled() {
+    let (glean, _t) = new_glean(None);
+    let metric = StringMetric::new(CommonMetricData {
+        category: "category".to_string(),
+        name: "string_metric".to_string(),
+        send_in_pings: vec!["baseline".to_string()],
+        ..Default::default()
+    });
+    let another_metric = StringMetric::new(CommonMetricData {
+        category: "category".to_string(),
+        name: "another_string_metric".to_string(),
+        send_in_pings: vec!["baseline".to_string()],
+        ..Default::default()
+    });
```

Comment:
Let's turn this into a labeled metric to test with:

```suggestion
    let another_metric = LabeledString::new(
        CommonMetricData {
            category: "category".to_string(),
            name: "another_string_metric".to_string(),
            send_in_pings: vec!["baseline".to_string()],
            ..Default::default()
        },
        Some(vec!["label1".into()]),
    );
```

Then use this below:

```rust
    another_metric.get("label1").set_sync(&glean, "TEST_VALUE");
    assert_eq!(
        "TEST_VALUE",
        another_metric
            .get("label1")
            .get_value(&glean, "baseline")
            .unwrap(),
        "Initial value must match"
    );
```